opentracing-mock/src/main/java/io/opentracing/mock/MockTracer.java

```diff
@@ -86,13 +85,39 @@ public synchronized void reset() {
     }
 
     /**
-     * @return a copy of all finish()ed MockSpans started by this MockTracer (since construction or the last call to
+     * @return all finish()ed MockSpans started by this MockTracer (since construction or the last call to
      * MockTracer.reset()).
      *
      * @see MockTracer#reset()
      */
     public synchronized List<MockSpan> finishedSpans() {
-        return new ArrayList<>(this.finishedSpans);
+        return Collections.unmodifiableList(new ArrayList<>(this.finishedSpans));
+    }
+
+    /**
+     * @return all finish()ed Traces(Spans) started by this MockTracer grouped by traceId and spanId in HashMap format.
+     */
+    public Map<String, Map<String, MockSpan>> finishedTraces() {
```

Comment:
I just noticed that method `finishedSpans` is **synchronized**.

May be a good idea to make `finishedTraces` synchronized as well?

Reply:
+1

Reply:
Done.